cmd/enroll.go

```diff
@@ -0,0 +1,103 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+type enrollTokenData struct {
+	Token string `json:"token"`
+}
+
+type enrollResponse struct {
+	Success bool            `json:"success"`
+	Message string          `json:"message"`
+	Data    enrollTokenData `json:"data"`
+}
+
+func getApiUrl(firstLetter string) (*string, error) {
+	apiUrls := map[string]string{
+		"P": "https://api.shopmonkey.cloud/",
+		"S": "https://sandbox-api.shopmonkey.cloud/",
+		"E": "https://edge-api.shopmonkey.cloud/",
+		"L": "http://localhost:3101",
+	}
+
+	if url, exists := apiUrls[firstLetter]; exists {
+		return &url, nil
+	}
+	return nil, fmt.Errorf("Invalid first letter")
+}
+
+var enrollCmd = &cobra.Command{
+	Use:   "enroll",
+	Short: "Enroll a new server and get api key",
+	Run: func(cmd *cobra.Command, args []string) {
+		logger := newLogger(cmd)
+		logger = logger.WithPrefix("[enroll]")
+		code := mustFlagString(cmd, "code", true)
+
+		firstLetter := code[0:1]
+		apiURL, err := getApiUrl(firstLetter)
```

Comment:
this is kinda silly but can you _also_ support providing a url via api-url flag so we can test this on previews

Reply:
Yep! Makes sense.